automod/rules/spammentions.go

```diff
@@ -0,0 +1,38 @@
+package rules
+
+import (
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/automod"
+)
+
+var _ automod.PostRuleFunc = SpamMentionsRule
+
+var mentionHourlyThreshold = 20
+
+// SpamMentionsRule looks for accounts which mention an unusually large number of distinct accounts per period.
+func SpamMentionsRule(evt *automod.RecordEvent, post *appbsky.FeedPost) error {
+	did := evt.Account.Identity.DID.String()
+
+	// Increment counters for all new mentions in this post.
+	var newMentions bool
+	for _, facet := range post.Facets {
+		for _, feature := range facet.Features {
+			mention := feature.RichtextFacet_Mention
+			if mention == nil {
+				continue
+			}
```

Comment:
there is an `ExtractFacets` helper function, though in this case the code looks pretty tight without it:
https://pkg.go.dev/github.com/bluesky-social/indigo/automod/rules#ExtractFacets